engines/prince/prince.cpp

```diff
@@ -54,6 +54,111 @@
 
 namespace Prince {
 
+#ifdef USE_TTS
+
+// Mazovia encoding
+static const int polishEncodingTable[] = {
+	0x86, 0xc485, 0x8d, 0xc487, 0x8f, 0xc484, 0x90, 0xc498,	// ą, ć, Ą, Ę
+	0x91, 0xc499, 0x92, 0xc582, 0x95, 0xc486, 0x98, 0xc59a,	// ę, ł, Ć, Ś
+	0x9c, 0xc581, 0x9e, 0xc59b, 0xa0, 0xc5b9, 0xa1, 0xc5bb,	// Ł, ś, Ź, Ż
+	0xa2, 0xc3b3, 0xa3, 0xc393, 0xa4, 0xc584, 0xa5, 0xc583,	// ó, Ó, ń, Ń
+	0xa6, 0xc5ba, 0xa7, 0xc5bc,								// ź, ż
+	0
+};
+
+// Custom encoding
+static const int russianEncodingTable[] = {
+	0x46, 0xd090, 0x66, 0xd0b0, 0x83, 0xd091, 0x92, 0xd0b1,	// А, а, Б, б
+	0x44, 0xd092, 0x64, 0xd0b2, 0x55, 0xd093, 0x75, 0xd0b3,	// В, в, Г, г
+	0x4c, 0xd094, 0x6c, 0xd0b4, 0x54, 0xd095, 0x74, 0xd0b5,	// Д, д, Е, е
+	0x81, 0xd096, 0x8f, 0xd0b6, 0x50, 0xd097, 0x70, 0xd0b7,	// Ж, ж, З, з
+	0x42, 0xd098, 0x62, 0xd0b8, 0x51, 0xd099, 0x71, 0xd0b9,	// И, и, Й, й
+	0x52, 0xd09a, 0x72, 0xd0ba, 0x4b, 0xd09b, 0x6b, 0xd0bb,	// К, к, Л, л
+	0x56, 0xd09c, 0x76, 0xd0bc, 0x59, 0xd09d, 0x79, 0xd0bd,	// М, м, Н, н
+	0x4a, 0xd09e, 0x6a, 0xd0be, 0x47, 0xd09f, 0x67, 0xd0bf,	// О, о, П, п
+	0x48, 0xd0a0, 0x68, 0xd180, 0x43, 0xd0a1, 0x63, 0xd181,	// Р, р, С, с
+	0x4e, 0xd0a2, 0x6e, 0xd182, 0x45, 0xd0a3, 0x65, 0xd183,	// Т, т, У, у
+	0x41, 0xd0a4, 0x61, 0xd184, 0x7f, 0xd0a5, 0x85, 0xd185,	// Ф, ф, Х, х
+	0x57, 0xd0a6, 0x77, 0xd186, 0x58, 0xd0a7, 0x78, 0xd187,	// Ц, ц, Ч, ч
+	0x49, 0xd0a8, 0x69, 0xd188, 0x4f, 0xd0a9, 0x6f, 0xd189,	// Ш, ш, Щ, щ
+	0x8d, 0xd18a, 0x53, 0xd0ab, 0x73, 0xd18b, 0x4d, 0xd0ac,	// ъ, Ы, ы, Ь
+	0x6d, 0xd18c, 0x82, 0xd0ad, 0x90, 0xd18d, 0x92, 0xd18e,	// ь, Э, э, ю
+	0x5a, 0xd0af, 0x7a, 0xd18f,								// Я, я
+	0
+};
+
+// Custom encoding
+static const int spanishEncodingTable[] = {
+	0x25, 0xc3ad, 0x26, 0xc3ba, 0x35, 0xc3a1, 0x36, 0xc3a9, // í, ú, á, é
+	0x37, 0xc2bf, 0x38, 0xc3b1, 0x3b, 0xc2a1, 0x5f, 0xc3b3, // ¿, ñ, ¡, ó
+	0
+};
+
+// Custom encoding
+static const int germanEncodingTable[] = {
+	0x83, 0xc384, 0x84, 0xc396, 0x85, 0xc39c, 0x7f, 0xc39f,  // Ä, Ö, Ü, ß
+	0x80, 0xc3a4, 0x81, 0xc3b6, 0x82, 0xc3bc,				 // ä, ö, ü
+	0
+};
+
+struct CharacterVoiceData {
+	uint32 textColor;
```

Comment:
I think this table could be reduced in size by using `uint8, uint8, uint8, int8, bool`